tools/java/org/unicode/cldr/api/FilteredData.java

```diff
@@ -0,0 +1,73 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package org.unicode.cldr.api;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Filters underlying {@link CldrData} source data to transform values or remove values on the fly.
+ * The order of transformed values produced by the class is the same order as the underlying data.
+ *
+ * <p>This class does not currently support injecting additional values due to the difficulty in
+ * ensuring correct ordering semantics.
+ *
+ * <p>A subclass should implement the {@link FilteredData#filter(CldrValue)} method to achieve the
+ * desired results. Note however that this API is currently unstable and is expected to change once
+ * DTD path ordering is enforced everywhere.
+ */
+// TODO: Once DTD ordering is the only allowed order, this can be extended to allow adding paths.
+public abstract class FilteredData implements CldrData {
+    private final CldrData src;
+
+    /** Constructs the parent class with a {@link CldrData} source. */
+    protected FilteredData(CldrData src) {
+        this.src = checkNotNull(src);
+    }
+
+    /** Returns the underlying source data to sub-class implementations. */
+    protected CldrData getSourceData() {
+        return src;
+    }
+
+    /**
+     * Returns a filtered CLDR value, replacing or removing the original value during visitation.
+     * The filtered value can only differ in it's base value or value attributes, and must have
+     * the same {@link CldrPath} associated with it.
+     *
+     * @return the filtered to be replaced, or {@code null} to remove the value.
+     */
+    protected abstract CldrValue filter(CldrValue value);
+
+    @Override
+    public final void accept(PathOrder order, ValueVisitor visitor) {
+        src.accept(order, v -> visitFiltered(v, visitor));
```

Comment:
I am not familiar with Java lambdas. I see that this is one, but can you say briefly what it does?
It seems to call the visitFiltered(value, visitor) method on line 59, and the return value would be passed into src.accept(order, return value) but visitFiltered() is void.
Sorry for my ignorance...

Reply:
The lambda is describing a ValueVisitor (because of the method it's being passed to). This has the method:
    void visit(CldrValue value)

So think of:
    v -> visitFiltered(v, visitor)

As describing the body of a visit(CldrValue) method that takes CldrValue 'v'.